test/provisioner/provision_libvirt.go

```diff
@@ -19,11 +19,18 @@ import (
 
 // LibvirtProvisioner implements the CloudProvisioner interface for Libvirt.
 type LibvirtProvisioner struct {
-	conn       *libvirt.Connect // Libvirt connection
-	network    string           // Network name
-	storage    string           // Storage pool name
-	wd         string           // libvirt's directory path on this repository
-	volumeName string           // Podvm volume name
+	conn         *libvirt.Connect // Libvirt connection
+	network      string           // Network name
+	ssh_key_file string           // SSH key file used to connect to Libvirt
+	storage      string           // Storage pool name
+	uri          string           // Libvirt URI
```

Comment:
I think the mixture between testing environment properties and CAA properties is a bit confusing, would it make sense to have something like a `caaCM map[string]string` field (& similarly caaSecret if needed), which will be mapped directly to the ENV_VARS & values required for the CAA? it would help to avoid changing the testing code itself with each new/modified variable CAA supports (it happened few times already for azure, vsphere and common vars were also added couple of times)
WDYT? is it even feasible?